virttest/qemu_vm.py

```diff
@@ -1340,6 +1340,10 @@ def sort_key(dev):
             for pcic in params.objects("pci_controllers"):
                 dev = devices.pcic_by_params(pcic, params.object_params(pcic))
                 pcics.append(dev)
+            extra_root_port = params.get('extra_root_port', '0')
+            for num in range(int(extra_root_port)):
+                pcics.append(devices.pcic_by_params("pcie_root_port_%s" % num,
```

Comment:
@vivianQizhu  could we use the unified bus id? 
extra root port "-device pcie-root-port,id=pcie_root_port_0,slot=2,chassis=2,addr=0x2,bus=pcie.0 \"
add root port: -device pcie-root-port,id=pcie.0-root-port-5,slot=5,chassis=5,addr=0x5,bus=pcie.0 \

Reply:
@jingzhao84 That's intended, since add_root_port() inside QPCIEBus class use address to define the id, and here it can't get addr so use index, better to give them different prefix to avoid the possible conflict.

Reply:
Yep, good idea. Also I'd make a soft-fail in case we fail to insert them in case no free port is available (to avoid the need to set "extra_root_port = 0" on places where all ports are used).

Reply:
Btw good name could be `pcie_extra_root_port` or `pcie_hotplug_root_port` to emphasize either the source or the usecase (I prefer the first one). I'd also consider `reserved` which already exists for reserved pci devices, but that would require changing all the related variables.